### PR TITLE
Remove AUTO-DRAFT in product name field on create new product page

### LINKED ISSUE #37930: Product Block Editor: Remove `AUTO-DRAFT` in product name field on create new product page
**Description**

As part of the create new product logic we have to set an initial name, which is currently set to `AUTO-DRAFT`.
This is being set in the name input field as well, which comes across a little confusing to a user.
We should display a blank value within the input field if the product is in auto draft (aka the name is set to `AUTO-DRAFT`).

Image

**Acceptance criteria**

- [ ] Show an empty product name field when name is `AUTO-DRAFT`.

### PR DESCRIPTION
**Pull request by mdperez86**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #37930

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Got to `/wp-admin/tools.php?page=woocommerce-admin-test-helper`
2. Under Features tab make sure to enable `product-block-editor`
3. Then visit `/wp-admin/admin.php?page=wc-admin&path=/add-product`
4. Under `General` tab and within `Basic details` section the product name should not longer be `AUTO-DRAFT` when creating the product.

### CODE REVIEW

File: packages/js/product-editor/src/blocks/name/edit.tsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ import {
 	Fragment,
 	createInterpolateElement,
 	useState,
+	useEffect,
```
Comment (louwie17) [inline packages/js/product-editor/src/blocks/name/edit.tsx]: Unnecessary import?

**Review by louwie17 — CHANGES_REQUESTED** [review]
Tested well, and code looks good, just looks like you added or left an unnecessary import `useEffect` (see inline comment).

File: packages/js/product-editor/src/blocks/name/edit.tsx (hunk 1 above)
Comment (mdperez86) [reply packages/js/product-editor/src/blocks/name/edit.tsx]: Thanks @louwie17. Done -> https://github.com/woocommerce/woocommerce/pull/37982/commits/1b453ebbf02049faf1af978a3535583b5f7e6f06

**Review by louwie17 — APPROVED** [review]
LGTM 🚀

**Pull request merged**